### PR TITLE
add anti ctec MVP

### PR DESCRIPTION
**Pull request by jukefr**
related to #4

### CODE REVIEW

File: src/main/java/io/github/coolmineman/cheaterdeleter/modules/packetcount/PacketLimiterCheck.java [hunk 1]
```diff
@@ -21,6 +21,22 @@ public PacketLimiterCheck() {
     public ActionResult onPacket(CDPlayer player, Packet<ServerPlayPacketListener> packet) {
         if (!enabledFor(player)) return ActionResult.PASS;
         PacketVolumeData data = player.getOrCreateData(PacketVolumeData.class, PacketVolumeData::new);
+
+        if (packet.getClass().getName().equals("net.minecraft.class_2840")) {
```
Comment (jukefr) [inline src/main/java/io/github/coolmineman/cheaterdeleter/modules/packetcount/PacketLimiterCheck.java]: definitely exists a far better way to do that

File: src/main/java/io/github/coolmineman/cheaterdeleter/modules/packetcount/PacketLimiterCheck.java [hunk 2]
```diff
@@ -21,6 +21,22 @@ public PacketLimiterCheck() {
...
+            if (data.craftingCount != 0 ) {
+                if (System.currentTimeMillis() - data.lastCraft < 125) {
```
Comment (jukefr) [inline src/main/java/io/github/coolmineman/cheaterdeleter/modules/packetcount/PacketLimiterCheck.java]: this is the only reason i can think of why the crasher doesnt get caught sometimes, other than a really inefficient looping of the limiter itself... (which I have no clue how to debug tbh)

File: src/main/java/io/github/coolmineman/cheaterdeleter/modules/packetcount/PacketLimiterCheck.java (hunk 1 above)
Comment (TheEpicBlock) [reply src/main/java/io/github/coolmineman/cheaterdeleter/modules/packetcount/PacketLimiterCheck.java]:
```java
if (packet instanceof CraftRequestC2SPacket) {
```
This is how this is usually done in networking

**Comment by jukefr** [comment]
also from more **anecdotal** testing it seems that the limiter will not catch the crasher when the player was afk for a while before or that he just joined the server as it started...
maybe its something to do with the way the data storage Class is instantiated ?

**Comment by jukefr** [comment]
latest version seems functional
had stupid logic going on 
maybe you want to move the check in its own module or something tho i'll leave that up to you

**Pull request closed without merging**